dynamiqs/sparse.py

```diff
@@ -73,28 +69,29 @@ def _sub_dia(self, other: SparseQArray) -> SparseQArray:
         out_offsets = tuple(sorted(out_offsets_diags.keys()))
         out_diags = jnp.array([out_offsets_diags[offset] for offset in out_offsets])
 
-        return SparseQArray(out_diags, out_offsets)
+        return SparseQArray(out_diags, out_offsets, self.dims)
+
+    def __rsub__(self, other: Array) -> Array:
+        return -self + other
 
     def __mul__(self, other: Array | SparseQArray) -> Array | SparseQArray:
         if isinstance(other, ScalarLike):
             if other == 0:
-                empty_diags = jnp.empty(0, self.shape[-1])
-                return SparseQArray(empty_diags, ())
-            return SparseQArray(other * self.diags, self.offsets)
+                diags = jnp.empty(0, self.shape[-1])
```

Comment:
@gautierronan we do not have a `shape` property in this PR, I think `self.shape[-1]` should instead be `self.dims[0]`.

Reply:
Yes, I guess this will be fixed once we have implemented the `.shape` method. Let's just wait for this I think.